Adding userAgent parameter to Isochrone request retrofit call URL

langsmith opened the pull request:
Resolves #996 by adding a required `/mapbox/` slug to the Isochrone API service's Retrofit request URL. Based on my demo app refactoring work in https://github.com/mapbox/mapbox-android-demo/pull/1029, the #996 bug renders the wrapper unusable. 

The API's docs

<img width="632" alt="Screen Shot 2019-04-11 at 3 57 45 PM" src="https://user-images.githubusercontent.com/4394910/55998305-9659b780-5c72-11e9-988f-46c4f0ff40fe.png">

Totally fine if this is the wrong way to fix the bug. Figured I'd open something to get the conversation started.

langsmith commented:
Actually, I'm now seeing how other navigation API wrappers in the repo, handle the user agent. Will adjust the isochrone support accordingly

osana commented:
@langsmith Technically speaking this becomes a 5.0.0 release because the API changed and it is not backward compatible. How come we did not catch this on the test level? I guess because a response was mocked. Did your sample code work?

langsmith commented:
> I am surprised the tests did not catch this? I guess because a response was mocked. Did your sample code work?

Yea, sample code worked. I think we didn't catch this because I put the `/mapbox/` slug in the `Makefile` isochrone fixture URLs (https://github.com/mapbox/mapbox-java/blob/master/Makefile#L199).

>Also, I think the way to fix this without making a major release would be to add the new method (not modify the existing one). One old one can be deprecated and we will remove it when 5.0.0 is done.

Sounds 💯 . I've pushed and rebased this fix. Added a new `getCall()` method which has the `user` parameter. Also added needed `deprecated` annotations to the previous `getCall()` method.

langsmith commented:
Going to test this work by using a Java services SNAPSHOT in the demo app isochrone refactor pr https://github.com/mapbox/mapbox-android-demo/pull/1029

osana commented on services-isochrone/src/main/java/com/mapbox/api/isochrone/MapboxIsochrone.java:
```diff
@@ -145,6 +150,16 @@ public static Builder builder() {
+     * @param user a non-null string which will replace the default user used in the Isochrone
+     *             request
+     * @return this builder for chaining options together
+     * @since 2.1.0
```
`@since` tag should be 4.7.0

langsmith replied:
So weird. How'd that happen :brainBoom:

langsmith replied:
Fixed

osana commented on services-isochrone/src/main/java/com/mapbox/api/isochrone/IsochroneCriteria.java:
```diff
@@ -13,6 +13,13 @@
+  /**
+   * Mapbox default username.
+   *
+   * @since 4.6.1
```
should be 4.7.0

langsmith replied:
Fixed

osana commented on services-isochrone/src/main/java/com/mapbox/api/isochrone/IsochroneService.java:
```diff
@@ -18,46 +18,102 @@
+   */
+  @GET("/isochrone/v1/{user}/{profile}/{coordinates}")
+  Call<FeatureCollection> getCall(
+    @Path("profile") String profile,
```
NIT: could we have the order as in the path: user, profile, coordinate

langsmith replied:
Fixed this, pushed, and rebased
